cmd/tg_bot/internal/handlers/handlers.go

```diff
@@ -1,25 +1,59 @@
 package handlers
 
 import (
+	"log"
+	"nodemon/cmd/tg_bot/internal"
+	"nodemon/cmd/tg_bot/internal/base_messages"
+
 	tele "gopkg.in/telebot.v3"
-	"nodemon/cmd/tg_bot/internal/messaging"
+	"nodemon/pkg/entities"
 )
 
-func InitHandlers(bot *tele.Bot, environment *messaging.MessageEnvironment) {
+func InitHandlers(bot *tele.Bot, environment *internal.TelegramBotEnvironment) {
 	bot.Handle("/hello", func(c tele.Context) error {
-		return c.Send("Hello!")
+		oldChatID, err := environment.ChatStorage.FindChatID(entities.TelegramPlatform)
+		if err != nil {
+			log.Printf("failed to insert chat id into db: %v", err)
+			return c.Send("An error occurred while finding the chat id in database")
+		}
+		if oldChatID != nil {
+			return c.Send("Hello! I remember this chat.")
+		}
+		chatID := entities.ChatID(c.Chat().ID)
+
+		err = environment.ChatStorage.InsertChatID(chatID, entities.TelegramPlatform)
+		if err != nil {
+			log.Printf("failed to insert chat id into db: %v", err)
+			return c.Send("I failed to save this chat id")
+		}
+		return c.Send("Hello! This new chat has been saved for alerting.")
```

Comment:
It doesn't look safe way. Anyone can interact with bot and subscribe for alerts.

Reply:
Yes, I am aware. Will be fixed in the next PR.